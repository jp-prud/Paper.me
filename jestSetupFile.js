import 'react-native-gesture-handler/jestSetup';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

export const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => {
      return {
        navigate: mockedNavigate,
        goBack: jest.fn(),
      };
    },
  };
});
