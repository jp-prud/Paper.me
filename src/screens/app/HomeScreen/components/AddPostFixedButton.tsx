import {ViewStyle} from 'react-native';

import {useNavigation} from '@react-navigation/native';

import {Icon, TouchableOpacityBox, TouchableOpacityBoxProps} from '@components';
import {$shadowProps} from '@theme';

export function AddPostFixedButton() {
  const {navigate} = useNavigation();

  function handlePressNavigate() {
    navigate('PreviewPostScreen');
  }

  return (
    <TouchableOpacityBox
      testID="add-post-fixed-button"
      onPress={handlePressNavigate}
      {...$touchableContainerStyles}
      style={[$shadowProps, $fixedContainerStyle]}>
      <Icon name="pencil" color="white" />
    </TouchableOpacityBox>
  );
}

const $touchableContainerStyles: TouchableOpacityBoxProps = {
  backgroundColor: 'primary',
  alignItems: 'center',
  justifyContent: 'center',
  width: 56,
  height: 56,
  borderRadius: 's32',
  borderWidth: 2,
  borderColor: 'teal200',
};

const $fixedContainerStyle: ViewStyle = {
  position: 'absolute',
  right: 24,
  bottom: 24,
};
