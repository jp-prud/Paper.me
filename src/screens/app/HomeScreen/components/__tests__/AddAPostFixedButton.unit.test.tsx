import {fireEvent, render} from '@tests';
import {mockedNavigate} from 'jestSetupFile.js';

import {AddPostFixedButton} from '../AddPostFixedButton';

describe('AddPostFixedButton - unit', () => {
  it('should render the component', () => {
    const {getByTestId, toJSON} = render(<AddPostFixedButton />);

    expect(getByTestId('add-post-fixed-button')).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });

  it("should navigate to 'PreviewPostScreen' when pressed", () => {
    const {getByTestId} = render(<AddPostFixedButton />);

    fireEvent.press(getByTestId('add-post-fixed-button'));

    expect(mockedNavigate).toHaveBeenCalledWith('PreviewPostScreen');
    expect(mockedNavigate).toHaveBeenCalledTimes(1);
  });
});
