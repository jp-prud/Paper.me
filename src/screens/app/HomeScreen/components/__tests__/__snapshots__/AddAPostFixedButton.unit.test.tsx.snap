// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPostFixedButton - unit should render the component 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#017582",
        "borderColor": "#c7e7ea",
        "borderRadius": 32,
        "borderWidth": 2,
        "bottom": 24,
        "elevation": 10,
        "height": 56,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 24,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.05,
        "shadowRadius": 12,
        "width": 56,
      }
    }
    testID="add-post-fixed-button"
  >
    <View
      style={
        [
          {},
        ]
      }
      testID="Icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M13.2601 3.59997L5.0501 12.29C4.7401 12.62 4.4401 13.27 4.3801 13.72L4.0101 16.96C3.8801 18.13 4.7201 18.93 5.8801 18.73L9.1001 18.18C9.5501 18.1 10.1801 17.77 10.4901 17.43L18.7001 8.73997C20.1201 7.23997 20.7601 5.52997 18.5501 3.43997C16.3501 1.36997 14.6801 2.09997 13.2601 3.59997Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
                "strokeMiterlimit",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeMiterlimit={10}
            strokeWidth="1.5"
          />
          <RNSVGPath
            d="M11.8899 5.04999C12.3199 7.80999 14.5599 9.91999 17.3399 10.2"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
                "strokeMiterlimit",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeMiterlimit={10}
            strokeWidth="1.5"
          />
          <RNSVGPath
            d="M3 22H21"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
                "strokeMiterlimit",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeMiterlimit={10}
            strokeWidth="1.5"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
